.changeset/wet-readers-join.md

```diff
@@ -0,0 +1,34 @@
+---
+'astro': minor
+---
+
+Shipped a new SSR mode, called `serverless`. 
+When enabled, Astro will emit a file for each page, which will render one single page.
+
+These files will be emitted inside `dist/pages`, and they will look like this:
+
+```
+├── pages
+│   ├── blog
+│   │   ├── entry._slug_.astro.mjs
+│   │   └── entry.about.astro.mjs
+│   └── entry.index.astro.mjs
+```
+
+To enable and customise this mode, new options are now available:
+
+```js
+export default defineConfig({
+    output: "server",
+    adapter: node({
+        mode: "standalone"
+    }),
+    build: {
+        ssrMode: "serverless",
+        serverlessEntryPrefix: "main"
+    }
+})
+```
+
+- `ssrMode` accepts two values, `"server"` or `"serverless"`. Default value, `"server"`. 
+- `serverlessEntryPrefix` allows to change the prefix of a serverless
```

Comment:
I wonder if `serverless` is the right term here. The previous build output can and are also used on serverless platforms. SvelteKit's Vercel adapter calls it ["split"](https://kit.svelte.dev/docs/adapter-vercel#deployment-configuration), since we're doing on the build-level, maybe we can call it `build.pageCodeSplit`? Vite has a similar option called `vite.build.cssCodeSplit`.

I don't really understand the `serverlessEntryPrefix` part yet, but maybe I will when I finish reviewing this PR!

Reply:
are you proposing `build.pageCodeSplit` be a boolean or what?

Reply:
Yeah, it would be a boolean which defaults to false.

Reply:
I probably misunderstand the meaning, because in my head it seems we split the code of a page in multiple files. 

But it makes sense to rename the option to something that's more familiar to the majority of users